inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/LearningCurveChartPanel.java

```diff
@@ -0,0 +1,242 @@
+     * <pre>
+     * recommender1: 'step', recommender2 : 'step'
+     * </pre>
+     * 
+     * @return
+     * @throws IOException
+     */
+    private String createJSScript(String aDataColumns, String aChartType) throws IOException
+    {
+        int[] intArray = IntStream.range(0, maxPointsToPlot).map(i -> i).toArray();
+        String xaxisValues = "[ 'x' ," + substring(Arrays.toString(intArray), 1, -1) + "]";
+        String data = toJsonString(aDataColumns).substring(1, aDataColumns.toString().length());
+
+        // bind data to chart container
+        String javascript = "var chart=c3.generate({bindto:'#" + chartContainer.getMarkupId()
+                + "',data:{ x:'x', columns:[" + xaxisValues + " ," + data + "],types:{" + aChartType
+                + "}}});;";
+        return javascript;
+    }
+
+    /**
+     * Fetches a number of latest evaluation scores from the database and save it in the map
+     * corresponding to each recommender for which the scores have been logged in the database
+     * 
+     * @return
+     */
+    private MultiValuedMap<String, Double> getLatestScores(RenderAnnotationsEvent aEvent)
+    {
+        // we want to plot RecommenderEvaluationResultEvent for the learning curve. The
+        // value of the event
+        String eventType = "RecommenderEvaluationResultEvent";
+
+        List<LoggedEvent> loggedEvents = eventRepo.listLoggedEvents(model.getObject().getProject(),
+                model.getObject().getUser().getUsername(), eventType, maxPointsToPlot);
+
+        if (CollectionUtils.isEmpty(loggedEvents)) {
+            return new ArrayListValuedHashMap<String, Double>();
+        }
+
+        // we want to show the latest record on the right side of the graph
+        Collections.reverse(loggedEvents);
+
+        MultiValuedMap<String, Double> recommenderScoreMap = new ArrayListValuedHashMap
+                <String, Double>();
+
+        // iterate over the logged events to extract the scores and map it against its corresponding
+        // recommender.
+        for (LoggedEvent loggedEvent : loggedEvents) {
+            String detailJson = loggedEvent.getDetails();
+            try {
+                Details detail = fromJsonString(Details.class, detailJson);
+
+                //do not include the scores from disabled recommenders
+                List<Recommender> recommenderIfActive = recommendationService.getEnabledRecommenders(model.getObject().getProject(), detail.layer, detail.tool);
+                if(recommenderIfActive.isEmpty())
+                {
+                    continue;
+                }
+
+                // sometimes score values NaN. Can result into error while rendering the graph on UI
+                if (!Double.isFinite(detail.score)) {
+                    continue;
+                }
+
+                recommenderScoreMap.put(detail.tool,
+                        Double.valueOf(new DecimalFormat("#.0000").format(detail.score)));
```

Comment:
This causes an exception on my machine: 

```
Caused by: java.lang.NumberFormatException: For input string: ",4444"
	at jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054) ~[?:?]
	at jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110) ~[?:?]
	at java.lang.Double.parseDouble(Double.java:543) ~[?:?]
	at java.lang.Double.valueOf(Double.java:506) ~[?:?]
	at de.tudarmstadt.ukp.inception.recommendation.sidebar.LearningCurveChartPanel.getLatestScores(LearningCurveChartPanel.java:225) ~[classes/:?]
	at de.tudarmstadt.ukp.inception.recommendation.sidebar.LearningCurveChartPanel.onRenderAnnotations(LearningCurveChartPanel.java:105) ~[classes/:?]
```

Reply:
Instead of handling this on the Java side, better pass the proper number to the client and handle it in JS (cf. https://github.com/c3js/c3/issues/2000#issuecomment-291787879).

Reply:
I will move it to the JS. But I don't understand how the input string ended up to be ",4444". It is the value coming from database and isn't it always a double value?

Reply:
If one doesn't pay close attention and always specifies an explicit format, then the float-to-string conversion uses the system locale. I guess with my system locale settings, floats are rendered with a comma instead of a dot by default.

Reply:
Resolved.